config/root-config.in

```diff
@@ -556,11 +556,11 @@ while test $# -gt 0; do
       out="$out @ROOT_VERSION@"
       ;;
     --git-revision)
-      ### Output the git revision number.  If RGitCommit.h can not be found, give up.
-      if test -r "${incdir}/RGitCommit.h"; then
-         out="$out `sed -n 's,.*ROOT_GIT_COMMIT *\"\(.*\)\".*,\1,p' < \"${incdir}/RGitCommit.h\"`"
+      ### Output the git revision.  If gitinfo.txt can not be found, use RVersion.hxx or give up.
```

Comment:
Why is the behavior of `--version` relevant to the behavior of `--git-revision` ? (Besides `--version` does not technically use `RVersion.hxx` but it source of information instead).

Currently the code (as far as I can tell) of `--git-revision` does "only"  use `gitinfo.txt` otherwise call `exit 1`.  What am I missing?

Reply:
I cannot follow. Let me add many words, maybe it helps. Given I cannot follow it might not help.

In a previous PR I forgot to fix the version part in root-config.bat.in, I am trying to sneak this in as part of this PR: [`4bdcf28` (#13653)](https://github.com/root-project/root/pull/13653/commits/4bdcf28f57fecd82309acb91125c9bf1e3f79188)

> Why is the behavior of `--version` relevant to the behavior of `--git-revision` ?

I don't know, I don't think it should be. Why do you ask?

> Besides `--version` does not technically use `RVersion.hxx` but it source of information instead

I don't think "but it source of information instead" is well formed, and while I could guess I don't know what to answer. This PR isn't about `RVersion.hxx`. That said, yes, the CMake variable `ROOT_VERSION` is determined from `RVersion.hxx` when the source directory is not a git repo. But again, I don't understand why we are discussing this here. I am completely lost with this discussion, Philippe - why you bring this up, what you want me to fix etc. :shrug:

> Currently the code (as far as I can tell) of `--git-revision` does "only"  use `gitinfo.txt` otherwise call `exit 1`.  What am I missing?

Nothing, that's correct. Nice, he? You're thus welcome to approve it :-) (I will be resolving the relnotes conflict and update `gitinfo.txt` on every build in the meantime.)

Reply:
Ooooh no I see what you're talking about - just the comment. Sorry, missed that, will fix. (Tbh I think this was caused by me tripping over your malformed initial sentence which I needed to interpret to make sense of and interpreted incorrectly...)

Reply:
> Tbh I think this was caused by me tripping over your malformed initial sentence 

Sorry about that :(